pyccel/parser/scope.py

```diff
@@ -366,12 +366,11 @@ def remove_variable(self, var, name = None):
                 Default : var.name.
         """
         if name is None:
-            name = self._original_symbol[var.name]
-
-        self._used_symbols.pop(name)
+            name = self.get_python_name(var.name)
 
         if name in self._locals['variables']:
             self._locals['variables'].pop(name)
+            self._used_symbols.pop(name)
```

Comment:
Could you please explain these changes? Was `name` set to be a `PyccelSymbol` instead of a string, when not provided to this function?

Reply:
`self._original_symbol` gets the Python name from exactly this scope. `self.get_python_name` gets the Python name from this scope or an enclosing scope. If we were trying to remove a variable from an enclosing scope (as happens if an inline method is called in a loop) then `self._original_symbol[var.name]` raises a `KeyError`. Once replaced by `self.get_python_name`, `self._used_symbols.pop(name)` raises the error instead which is why it was moved to where we know we are in the correct scope